src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/GenericStatus.java

```diff
@@ -73,5 +73,22 @@
     /**
      * Not complete: we are waiting for user input to continue (special case of IN_PROGRESS)
      */
-    PAUSED_PENDING_INPUT
+    PAUSED_PENDING_INPUT;
+
+    /**
+     * Create a {@link GenericStatus} from a {@link Result}
+     */
+    public static GenericStatus fromResult(Result result) {
+        if (result == Result.NOT_BUILT) {
+            return GenericStatus.NOT_EXECUTED;
+        } else if (result == Result.ABORTED) {
+            return GenericStatus.ABORTED;
+        } else if (result == Result.FAILURE ) {
+            return GenericStatus.FAILURE;
+        } else if (result == Result.UNSTABLE ) {
+            return GenericStatus.UNSTABLE;
+        } else {
+            return GenericStatus.SUCCESS;
```

Comment:
Unsafe. Compare explicitly to `SUCCESS`, and fall back to maybe `NOT_EXECUTED` for unknown values.

(Yes I know you are just moving existing code around here.)

Reply:
@jglick this should be addressed now. Could you take another look?